server/api/services/users.service.js

```diff
@@ -129,7 +130,8 @@ class UsersService {
         .find({ role: 'user', totalScore: { $gt: user.totalScore } })
         .countDocuments();
 
-      return { user, rank: rank + 1 };
+      const achievement = await achievementModel.find({ userId: roll }).populate('eventId');
+      return { user, rank: rank + 1, achievement };
```

Comment:
We would also have to receive the `achievement` object in the controller from where this function is being called and send it back as part of the response.